### PR TITLE
add database finalizers add and remove logic

### PR DESCRIPTION
**Pull request by pinikomarov**
continue from : https://github.com/openstack-k8s-operators/lib-common/pull/54

### CODE REVIEW

**Comment by pinikomarov** [comment]
@stuggi @gibizer This is a fresh continuation of #54 

First question: How to push squashed commits on the same pr , I have done git rebase HEAD~4 (squashing 4 into one) , but that remains on my local branch how do I push that squash history into my origin (lib_common_pkomarov repo branch)

Secondly :
The code works ..

```sh
[r@r placement-operator_pkomarov]$  oc get MariaDBDatabase -o json|jq .items[].metadata.finalizers
null
[
  "PlacementAPI"
]
[r@r placement-operator_pkomarov]$ oc delete MariaDBDatabase placement
mariadbdatabase.mariadb.openstack.org "placement" deleted
[r@r placement-operator_pkomarov]$ oc get MariaDBDatabase placement
NAME        AGE
placement   38m
[r@r placement-operator_pkomarov]$ oc delete PlacementAPI placement
placementapi.placement.openstack.org "placement" deleted
[r@r placement-operator_pkomarov]$  oc get PlacementAPI placement

```  

 I've done some work  to fix your comments , please take a look ..

**Comment by stuggi** [comment]
> @stuggi @gibizer This is a fresh continuation of #54
> 
> First question: How to push squashed commits on the same pr , I have done git rebase -i HEAD~4 (squashing 4 into one) , but that remains on my local branch how do I push that squash history into my origin (lib_common_pkomarov repo branch)

you'd just have to push it to your github, like with new commits, just force push it.

File: modules/database/funcs.go [hunk 1]
```diff
@@ -229,3 +232,36 @@ func (d *Database) getDBWithName(
...
+
+//
+// Returns a *Database object with specified name and namespace
+//
+func DatabaseName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+) (*Database, error) {
+	// create a Database here by only filling what we need (name and namespace)
```
Comment (stuggi) [inline modules/database/funcs.go:244]: you don't set the namespace as mentioned in the comment. should the comment be updated or the namespace added bellow?

Comment (pinikomarov) [reply modules/database/funcs.go]: ok, 
changed it to "	// create a Database by suppplying a resource name"

**Comment by gibizer** [comment]
> @stuggi @gibizer This is a fresh continuation of #54
> 
> First question: How to push squashed commits on the same pr , I have done git rebase -i HEAD~4 (squashing 4 into one) , but that remains on my local branch how do I push that squash history into my origin (lib_common_pkomarov repo branch)

`$ git push <remote-repo-name> <localBranchName>:<remoteBranchName>`

If you always push localBranchName to remoteBranchName then you can use `--set-upstream` to persist the connection between the local branch and the remote branch.

The rebase changed the history of your local branch compared to the remote branch so you need `--force` to allow updating such history.

Here is my workflow:
1. `git checkout -b pr-xxx`
2. do some development
3. do one ore more `git commit`
4. `git push gibi` Where `gibi` points to repo I forked from lib-common to github.com/gibizer/lib-common. This will create a the `pr-xxx` branch in my fork. Also git will automatically set the local pr-xxx to track the pr-xxx in the gibi remote. (This is what you can do manually with `--set-upstream`)
5. create a pull request between gibi:pr-xxx and the upstream lib-common:master via github UI
6. Then if I get comments where I need to change an existing commit in the PR then I do `git rebase -i` and edit the commits as requested. This might mean squashing things as well. But the important bit is that it changes the history of already published commits on my local pr-xxx branch.
7. When I'm done fixing the comments locally I push the local branch to my remote: `git push gibi --force`. Github will automatically update the content of the pull request according to the pushed commits.

File: modules/database/funcs.go [hunk 2]
```diff
@@ -229,3 +232,36 @@ func (d *Database) getDBWithName(
 
 	return nil
 }
+
+//
+// Returns a *Database object with specified name and namespace
+//
+func DatabaseName(
```
Comment (gibizer) [inline modules/database/funcs.go:239]: I liked the previous `GetDatabaseByName` name for this function.

Comment (pinikomarov) [reply modules/database/funcs.go:239]: ok I'll change it to GetDatabaseByName

Comment (pinikomarov) [reply modules/database/funcs.go:239]: also thanks for the git explanation , some of it I was less sure of

**Comment by pinikomarov** [comment]
@stuggi  @stuggi does this patch need a lgtm from me ? or a  ok-to-test from you ?

**Review by gibizer — APPROVED** [review]
looks good

**Review by stuggi — APPROVED** [review]
/lgtm

File: modules/database/funcs.go [hunk 3]
```diff
@@ -229,3 +232,36 @@ func (d *Database) getDBWithName(
 
 	return nil
 }
+
+//
+// Returns a *Database object with specified name and namespace
```
Comment (stuggi) [inline modules/database/funcs.go:237]: comment needs to start with GetDatabaseByName

File: modules/database/funcs.go [hunk 4]
```diff
@@ -229,3 +232,36 @@ func (d *Database) getDBWithName(
...
+	}
+	// then querying the MariaDBDatabase and store it in db by calling
+	if err := db.getDBWithName(ctx, h); err != nil {
+		return db, err
+	}
+	return db, nil
+}
+
+//
+// Delete a finalizer by its object
```
Comment (stuggi) [inline modules/database/funcs.go:256]: comment needs to start with DeleteFinalizer to make lint happy

Comment (pinikomarov) [reply modules/database/funcs.go]: done

File: modules/database/funcs.go (hunk 3 above)
Comment (pinikomarov) [reply modules/database/funcs.go]: done

**Comment by pinikomarov** [comment]
@stuggi Fixed the lint issues , can you add oktotest so it gitworflows can check it ?

**Pull request merged**